### DISCUSSION TITLE
appimagetool vs appimage-builder

### DISCUSSION THREAD

**Discussion by kuiperzone** [AppImage/AppImageKit#1245] [Q&A] [answered]
appimagetool is provided as a tool by this repository.  See: https://appimage.github.io/appimagetool/

whereas appimage-builder seems to be the defacto builder tool: https://docs.appimage.org/packaging-guide/overview.html

What's the difference between them? Different projects which do the same job? Related? Not related?

**Comment by TheAssassin** [comment, marked as answer]
https://docs.appimage.org/packaging-guide/from-source/index.html#ref-packaging-from-source mentions tools like linuxdeploy, which can be used to create AppDirs (and AppImages subsequently), too. appimage-builder is not the "de facto tool".

appimagetool is used by linuxdeploy as well as appimage-builder, electron-builder and many other tools. All it does is take an AppDir and build the resulting AppImage. It's the last tool in the pipeline and requires a ready-to-package AppDir.

linuxdeploy, appimage-builder, electron-builder and all the others are used to bundle an application and its dependencies in an AppDir, too. They typically bundle appimagetool to then build the AppImage from the final AppDir.

Fun fact: linuxdeploy calls itself an "AppDir maintenance tool" for a reason. AppImage is just one possible output format it can use.

**Reply by kuiperzone** [reply]
Brilliant. Thanks
